bindings/python/crocoddyl/core/controls/poly-one.cpp

```diff
@@ -0,0 +1,78 @@
+                                                        double, const Eigen::Ref<const Eigen::VectorXd>&) const>(
+          "params", &ControlParametrizationModelPolyOne::params, bp::args("self", "data", "t", "u"),
+          "Compute the control parameters.\n\n"
+          ":param data: the data on which the method operates.\n"
+          ":param t: normalized time in [0, 1].\n"
+          ":param u: control value (dim control.nu).")
+      .def("convert_bounds", &ControlParametrizationModelPolyOne::convert_bounds, bp::args("self", "u_lb", "u_ub"),
+           "Convert the bounds on the control to bounds on the control parameters.\n\n"
+           ":param u_lb: lower bounds on u (dim control.nu).\n"
+           ":param u_ub: upper bounds on u (dim control.nu).\n"
+           ":return p_lb, p_ub: lower and upper bounds on the control parameters (dim control.np).")
+      .def<void (ControlParametrizationModelPolyOne::*)(const boost::shared_ptr<ControlParametrizationDataAbstract>&,
+                                                        double, const Eigen::Ref<const Eigen::VectorXd>&) const>(
+          "calcDiff", &ControlParametrizationModelPolyOne::calcDiff, bp::args("self", "data", "t", "p"),
+          "Compute the derivative of the control with respect to the parameters.\n\n"
+          ":param data: the data on which the method operates.\n"
+          ":param t: normalized time in [0, 1].\n"
+          ":param p: control parameters (dim control.np).")
+      .def("multiplyByJacobian", &ControlParametrizationModelPolyOne::multiplyByJacobian_J,
+           bp::args("self", "t", "p", "A"),
+           "Compute the product between the given matrix A and the derivative of the control with respect to the "
+           "parameters.\n\n"
+           ":param t: normalized time in [0, 1].\n"
+           ":param p: control parameters (dim control.np).\n"
+           ":param A: matrix to multiply (dim na x control.nu).\n"
+           ":return Product between A and the partial derivative of the value function (dim na x control.np).")
+      .def(
+          "multiplyJacobianTransposeBy", &ControlParametrizationModelPolyOne::multiplyJacobianTransposeBy_J,
+          bp::args("self", "t", "p", "A"),
+          "Compute the product between the transpose of the derivative of the control with respect to the parameters\n"
+          "and a given matrix A.\n\n"
+          ":param t: normalized time in [0, 1].\n"
+          ":param p: control parameters (dim control.np).\n"
+          ":param A: matrix to multiply (dim control.nu x na).\n"
+          ":return Product between the partial derivative of the value function (transposed) and A (dim control.np x "
+          "na).")
+      .add_property("nu", bp::make_function(&ControlParametrizationModelPolyOne::get_nu), "dimension of control tuple")
+      .add_property("np", bp::make_function(&ControlParametrizationModelPolyOne::get_np),
+                    "dimension of the control parameters");
+}
+
```

Comment:
We should create an specialized data structure and bind it here. See a more elaborated discussion, which motivates the idea of having specialized data structure, addressed in a comment below that I have partially copied here:

> The time-dependent coefficients of each control class should be stored in a specialized data, which in this case, the coefficients are .... Furthermore, I propose a standard name for these terms, ....
>
> **Why?**
> Crocoddyl API assumes that `calcDiff` is computed from any stored results in `calc`. Then, we are doing so if we stored this computed coefficient in `calc` and then reuse it in `calcDiff`. We can also introduce the same notion for `multiplyByJacobian` and `multiplyJacobianTransposeBy` ... .
>
> This also leads to the question: should we remove t from the `calc`, `calcDiff`, `multiplyByJacobian` and `multiplyJacobianTransposeBy`?. My personal view is a big YES as `t` can be seem as a model parameter rather than input. ...

Please do not response to this copied message here, use the discussion below :)

Reply:
I won't mark the need of creating and binding specialized data structures in the other cases. But be aware, that we need to do ti as well :)